polars/polars-arrow/src/kernels/rolling/no_nulls/quantile.rs

```diff
@@ -251,35 +245,91 @@ where
     ))
 }
 
+#[inline]
+fn compute_wq<T>(buf: &[(T, f64)], p: f64, wsum: f64, interp: QuantileInterpolOptions) -> T
+where
+    T: Debug
+        + NativeType
+        + Mul<Output = T>
+        + Sub<Output = T>
+        + NumCast
+        + ToPrimitive
+        + Zero
+        + IsFloat
+        + PartialOrd,
+{
+    // There are a few ways to compute a weighted quantile but no "canonical" way.
+    // This is mostly taken from the Julia implementation which was readable and reasonable
+    // https://juliastats.org/StatsBase.jl/stable/scalarstats/#Quantile-and-Related-Functions-1
+    let (mut s, mut s_old, mut vk, mut v_old) = (0.0, 0.0, T::zero(), T::zero());
+
+    // Once the cumulative weight crosses h, we've found our ind{ex/ices}. The definition may look
+    // odd but it's the equivalent of taking h = p * (n - 1) + 1 if your data is indexed from 1.
+    let h: f64 = p * (wsum - buf[0].1) + buf[0].1;
+    for &(v, w) in buf.iter().filter(|(_, w)| *w != 0.0) {
```

Comment:
Is the result also correct if we don't branch on `w == 0`? This might be faster even when there are `0.0` values because of pipelining.